src/compiler/parser.ts

```diff
@@ -1114,8 +1114,8 @@ namespace ts {
             return array;
         }
 
-        function finishNode<T extends Node>(node: T, end?: number): T {
-            node.end = end === undefined ? scanner.getStartPos() : end;

```

Comment:
we use this all the time. the new code will result in two statements. please run perf tests to ensure we are not messing up the engine optimizations.